### PR TITLE
Add SEntry

### PR DESCRIPTION
**Pull request by wazolab**

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -1,2 +1,4 @@
 API=https://host.docker.internal:9000
 FRONTEND_PORT=3000
+NUXT_PUBLIC_SENTRY_DSN="FILL THIS OUT"
```
Comment (frodrigo) [inline .env.template:3]: Sentry is optional, should be empty by default.

File: README.md [hunk 2]
```diff
@@ -47,6 +47,15 @@ npx @eslint/config-inspector
 ```
 Visit http://localhost:7777/ to view and play with your ESLint config
 
+### Setup SEntry
```
Comment (frodrigo) [inline README.md:50]:
```suggestion
### Setup Sentry
```

File: .env.template [hunk 3]
```diff
@@ -1,2 +1,4 @@
 API=https://host.docker.internal:9000
 FRONTEND_PORT=3000
+NUXT_PUBLIC_SENTRY_DSN=
+NUXT_PUBLIC_SENTRY_ENVIRONMENT=development
```
Comment (frodrigo) [inline .env.template:4]: Why we need to set explicitly `development` ? Not better to set nothging or `prod` ?

File: nuxt.config.ts [hunk 4]
```diff
@@ -24,6 +24,10 @@ export default defineNuxtConfig({
   runtimeConfig: {
     public: {
       API: process.env.API,
+      sentry: {
+        dsn: '',
```
Comment (frodrigo) [inline nuxt.config.ts:28]: ?
```suggestion
        dsn: undefined,
```

File: plugins/sentry.client.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      Sentry.replayIntegration(),
+    ],
+
+    // Set tracesSampleRate to 1.0 to capture 100%
+    // of transactions for performance monitoring.
+    // We recommend adjusting this value in production
+    tracesSampleRate: 1.0, // Change in prod
+
+    // Set `tracePropagationTargets` to control for which URLs distributed tracing should be enabled
+    tracePropagationTargets: ['localhost', 'https://your-server.com'],
```
Comment (frodrigo) [inline plugins/sentry.client.ts:26]: Should be the remote API URL (or just skip this option)

File: plugins/sentry.client.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    environment: sentry.environment,
+    integrations: [
+      Sentry.browserTracingIntegration({ router }),
+      Sentry.replayIntegration(),
+    ],
+
+    // Set tracesSampleRate to 1.0 to capture 100%
+    // of transactions for performance monitoring.
+    // We recommend adjusting this value in production
+    tracesSampleRate: 1.0, // Change in prod
```
Comment (frodrigo) [inline plugins/sentry.client.ts:23]: We are in prod.

```suggestion
    tracesSampleRate: 0.05,
```

File: plugins/sentry.client.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+    // Set `tracePropagationTargets` to control for which URLs distributed tracing should be enabled
+    tracePropagationTargets: ['localhost', 'https://your-server.com'],
+
+    // Capture Replay for 10% of all sessions,
+    // plus for 100% of sessions with an error
+    replaysSessionSampleRate: 0.1, // Change in prod
+    replaysOnErrorSampleRate: 1.0, // Change in prod if necessary
```
Comment (frodrigo) [inline plugins/sentry.client.ts:31]:
```suggestion
    replaysSessionSampleRate: 0.05,
    replaysOnErrorSampleRate: 1.0,
```

File: .env.template (hunk 3 above)
Comment (wazolab) [reply .env.template:4]: We could set nothing as we already have a default case in `nuxt.config.ts` at `runtimeConfig.public.sentry.environment`.
But this is the `.env.template` so I think it's fine to propose all the keys supported by the app no ?

File: plugins/sentry.client.ts (hunk 6 above)
Comment (wazolab) [reply plugins/sentry.client.ts:23]: You mean 0.5 (50% of errors sent) no ?

File: .env.template (hunk 3 above)
Comment (frodrigo) [reply .env.template:4]: The `.env.template` should work as is when copied as `.env`

Comment (wazolab) [reply .env.template:4]: Isn't it the case ?
When in dev (it's superfluous but works) and in prod we just have to change the value of `NUXT_PUBLIC_SENTRY_ENVIRONMENT` to `production`.

But we could set it to `production` directly if you prefer ?

File: plugins/sentry.client.ts (hunk 6 above)
Comment (frodrigo) [reply plugins/sentry.client.ts]: It is for  "performance monitoring" not errors.

File: README.md [hunk 8]
```diff
@@ -47,6 +47,15 @@ npx @eslint/config-inspector
 ```
 Visit http://localhost:7777/ to view and play with your ESLint config
 
+### Setup Sentry
+
+Add DSN key and environment in .env file.
+
+```bash
+NUXT_PUBLIC_SENTRY_DSN="FILL THIS OUT"
+NUXT_PUBLIC_SENTRY_ENVIRONMENT=development
```
Comment (frodrigo) [inline README.md:56]:
```suggestion
NUXT_PUBLIC_SENTRY_ENVIRONMENT=production
```

File: nuxt.config.ts [hunk 9]
```diff
@@ -24,6 +24,10 @@ export default defineNuxtConfig({
   runtimeConfig: {
     public: {
       API: process.env.API,
+      sentry: {
+        dsn: undefined,
+        environment: 'development',
```
Comment (frodrigo) [inline nuxt.config.ts:29]:
```suggestion
        environment: 'production',
```

**Pull request merged**